lib/hca/soap_parser.rb

```diff
@@ -15,19 +15,23 @@ class ValidationError < StandardError
     def on_complete(env)
       super
     rescue Common::Client::Errors::HTTPError => e
-      doc = parse_doc(env.body)
-      el = doc.locate(FAULT_STRING_EL)[0]
+      if env.status.to_i == 503
```

Comment:
When `env.status.to_i` is `503`, the `env.body` is not proper XML (it is '<html><body>No Server Available</body></html>') and trying to parse it causes an exception (line 21), so it should be caught here and treated as a timeout.

Reply:
maybe a separate instance of soap_parser though -- this middleware is used in a lot of places, its generic, so if you need special handling it should happen in an application specific middleware after this one or in lieu of this one.

Reply:
@saneshark this _is_ a separate instance of soap_parser.  `HCA::SOAPParser`